### PR TITLE
test: patch index parsing

### PR DESCRIPTION
**Pull request by Alizter**
We add some tests for patch_index_range parsing. Making sure that obvious edge cases are handled correctly.

### CODE REVIEW

File: src/commands/patch_index_range.rs [hunk 1]
```diff
@@ -7,6 +7,12 @@ pub struct PatchIndexRange {
     pub end_index: Option<usize>,
 }
 
+impl PartialEq for PatchIndexRange {
+    fn eq(&self, other: &Self) -> bool {
+        self.start_index == other.start_index && self.end_index == other.end_index
+    }
+}
+
```
Comment (Alizter) [inline src/commands/patch_index_range.rs:15]: One thing I am unsure of is if `1-1` should get parsed as `1`. If that's the case then we will need to update this to equate equal ranges, or at least never produce them.

**Review by drewdeponte — APPROVED** [review]

**Comment by drewdeponte** [comment]
Manually signed off and integrated this, so I am closing this PR.

File: src/commands/patch_index_range.rs [hunk 2]
```diff
@@ -61,3 +67,151 @@ impl std::str::FromStr for PatchIndexRange {
...
+    );
+}
+
+// CR-someday alizter: We don't accept reflexive ranges, maybe we should?
+// It seems simple enought to parse this as a single patch index.
+#[test]
+fn range3() {
+    let patch_index_or_range = "2-2".parse::<PatchIndexRange>();
+    assert!(patch_index_or_range.is_err());
+}
```
[the remark this answers is not in the document]
Comment (drewdeponte) [reply src/commands/patch_index_range.rs:173]: yeah, I changed that test assertion.
